eks.go

```diff
@@ -0,0 +1,106 @@
+/******************************************************************************
+Cloud Resource Counter
+File: eks.go
+
+Summary: Provides a count of all EKS nodes.
+******************************************************************************/
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/eks"
+	color "github.com/logrusorgru/aurora"
+)
+
+// EKSNodes retrieves the count of all EKS Nodes either for all
+// regions (allRegions is true) or the region associated with the
+// session. This method gives status back to the user via the supplied
+// ActivityMonitor instance.
+func EKSNodes(sf ServiceFactory, am ActivityMonitor, allRegions bool) int {
+	nodeCount := 0
+
+	errs := make([]string, 0)
```

Comment:
**Thought _(non-blocking)_:**
Not gonna lie, I think the way you're passing the errs string slice is pretty cool & it enables the shorthand `+=` assignment from the function calls which is sleek.

But, I'd say that if a function errors in Go, you'd expect it to be returned as the last returned value of that func, not appended to a dereferenced string slice. So though it might read a bit more clunky, I would follow that pattern and you'd have several more lines to iterate the nodeCount.

```
errs := make([]error, 0)

count, err := eksCountForSingleRegion(regionName, sf, am)
if err != nil {
  errs = append(errs, err)
} else {
  nodeCount += count
}
```

Reply:
```
for _, err := range errs {
  am.SubResourceError(err.Error())
}
```

Reply:
for this part, i get what you're saying but the idea was to allow the code to run even if there are errors. so for this inner loop, how would you suggest tracking the error but allowing the loop to continue? i'm not sure there's a way for me to bubble up the error from inside this function?
https://github.com/expel-io/aws-resource-counter/blob/6d85f7c2a25a7bbc408d4aef3cf3b0c032a453a9/eks.go#L68-L96

Reply:
Oh, I see -- you could return (_, []error)
You'd have to keep a local var `errs` inside of the function to append as you go. Then you can bubble that array up @ the end.

```
count, eksErrs := eksCountForSingleRegion(regionName, sf, am)
if err != nil {
  errs = append(errs, eksErrs...)
} else {
  nodeCount += count
}
```

Reply:
Here's an example of how it could keep the errors in memory before returning upstream:
https://github.com/expel-io/aws-resource-counter/compare/la.MET-328...em.suggestion?expand=1

Reply:
I added a second commit to dry up some of the logic since my suggestion adds more duplicated lines.